src/Controls/src/Core/Handlers/Items2/iOS/CarouselViewController2.cs

```diff
@@ -132,12 +133,39 @@ private protected override async void AttachingToWindow()
 		{
 			base.AttachingToWindow();
 			Setup(ItemsView);
```

Comment:
When `Setup(ItemsView)` is called, it reconnects event handlers that can trigger UICollectionView's layout recalculation. At that exact moment, `RefreshVisibleItems()` queries `IndexPathsForVisibleItems`, which may return empty or stale data during the layout transition.

Can you review changing the order?
```
if (_wasDetachedFromWindow)
{
     RefreshVisibleItems(); // Query BEFORE layout changes
}
Setup(ItemsView); // Trigger layout AFTER we capture visible items
```